tasks/check_primary_cert.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+declare PT__installdir
+# shellcheck disable=SC1090
+source "$PT__installdir/ca_extend/files/common.sh"
+PUPPET_BIN='/opt/puppetlabs/puppet/bin'
+
+expiry_date="$($PUPPET_BIN/openssl x509 -enddate -noout -in "$($PUPPET_BIN/puppet config print hostcert)")" || {
+  fail "Error finding primary server certificate."
+}
+expiry_date="${expiry_date#*=}"
+expiry_seconds="$(date --date="$expiry_date" +"%s")" || fail "Error calculating expiry date from enddate"
+
+if (( $(date +"%s") >= expiry_seconds )); then
```

Comment:
How about an else case here that warns if it is going to expire in the next 3 months. Users could be running this proactively and it has yet to have expired but may not know about the regen option.

Reply:
Added.